pallets/parachain-system/src/lib.rs

```diff
@@ -480,10 +481,13 @@ impl<T: Config> Module<T> {
 	///
 	/// Checks if the sequence of the messages is valid, dispatches them and communicates the number
 	/// of processed messages to the collator via a storage update.
+	///
+	/// **Panics** if it turns out that after processing all messages the Message Queue Chain hash
+	///            doesn't match the expected.
```

Comment:
maybe better to panic in the call itself and return a result here, but anyway it is ok to me.

Reply:
Yeah, I thought about that but that doesn't feel to give lot of benifit. I am not against the change and happy to make it if you insist though.